File: backend/internal/models/invite_code.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type InviteCode struct {
	ID        uuid.UUID `gorm:"column:id;primaryKey;type:uuid;default:gen_random_uuid()" json:"id"`
	Code      string    `gorm:"column:code" json:"code"`
	CreatedBy uuid.UUID `gorm:"column:created_by" json:"created_by"`
	Uses      int       `gorm:"column:uses" json:"uses"`
	MaxUses   int       `gorm:"column:max_uses" json:"max_uses"`
	CreatedAt time.Time `gorm:"column:created_at" json:"created_at"`
}

File: frontend/app/types/inviteCode.ts
export interface InviteCode
{
	id         : string,
	code       : string,
	uses       : number,
	max_uses   : number,
	created_at : string
}

export interface FormattedInviteCode
{
	id        : string,
	code      : string,
	uses      : number,
	maxUses   : number,
	createdAt : string
}

export interface AddInviteCode
{
	code    : string | undefined,
	maxUses : string | undefined
}